lesson0.md

```diff
@@ -1,60 +1,106 @@
+
+> "When users don't control the program, we call it 'nonfree' or 'proprietary' software. The nonfree program controls the users, and the developer controls the program; this makes the program an instrument of unjust power."
+> - Richard Stallman, [The Free Software Definition](https://www.gnu.org/philosophy/free-sw.html)
+
+This marked the birth of **Free Software**, by which software is considered free if it respects the 4 fundamental freedoms:
+
+1. Anyone has the freedom to run the software as they wish, for any purpose
+2. Anyone has the freedom to study how the program works, and change it as they wish
+3. Anyone has the freedom to redistribute the software
+4. Anyone has the freedom to distribute copies of their own modified version of the Software
+
+As a consequence, access to the code must be open and all 3rd party software needed to run it must also comply with the 4 fundamental freedoms.
+
+A few years later, in 1997, Eric Raymond wrote an essay called **"The Cathedral and the Bazaar"**, in which he described how the Linux operating system and the Free Software philosophy had created a surprisingly effective and collaborative way of writing software.
+
+This set the stage for the appearance, one year later, of the term **Open Source** software. Open Source software is a close relative of Free Software but with a stronger focus on the methodology rather than the philosophy.
+
+The movement has continued growing since then and nowadays there are many examples of popular Open Source software such as, for example, WordPress, Firefox, React, Python, the Apache web server and the MySQL database.
+
+> **Exercise:** Find out more examples of Open Source software.
+>
+> **Exercise:** Can you think of any software that is "free" (as in you pay £0.00 for it) but would not qualify as Open Source?
+
+If you wish to find out more on this topic, please watch the excellent documentary ["Revolution OS"](https://www.youtube.com/watch?v=4vW62KqKJ5A) on YouTube.
+
+
+This section has sourced information from:
+* [GNU - Free Software Definition](https://www.gnu.org/philosophy/free-sw.html)
+* [Make WordPress Training - What is Open Source](https://make.wordpress.org/training/handbook/user-lessons/what-is-open-source/)
+* [Opensource.com - What's the difference between open source software and free software?](https://opensource.com/article/17/11/open-source-or-free-software)
+
 
 ## What is WordPress? WordPress.com versus WordPress.org
 
-### What is WordPress
+WordPress is an Open Source Content Management System (CMS), released in 2003 by Matt Mullenberg and Mike Little. As of 2017, it has 59% of the CMS marketshare and 28% of all websites, worldwide.
+
+Although WordPress started as a blogging platform, it can also be used for building websites (large and small), e-commerce sites (with the WooCommerce plugin), private community platforms (with the BuddyPress plugin), forums (with the bbPress plugin), etc.
```

Comment:
There are lots of ways to build eCommerce sites, community platforms and forums, with other plugins or with no third-party plugins. I don't know if we should be giving the impression that there are set plugins for specific functionality right off the bat.

Maybe instead we could say:

> WordPress started as a blogging platform. But today, it is used for building all kinds of large and small websites. You will find Open Source plugins that add features for eCommerce shops, media websites, private community platforms, forums, and more.